src/lib.rs

```diff
@@ -283,124 +179,84 @@ impl Renderer {
-            CPUAccessFlags: D3D11_CPU_ACCESS_WRITE,
+            BindFlags: 2,
+            CPUAccessFlags: 65536,
             MiscFlags: 0,
             StructureByteStride: 0,
         };
-        com_ptr_from_fn(|index_buffer| device.CreateBuffer(&desc, ptr::null(), index_buffer))
-            .map(|ib| Buffer(ib, len))
+
+        device.CreateBuffer(&desc, null())
+            .map_or(Ok(Buffer(None, len)), |buf| { Ok(Buffer(Some(buf), len)) })
     }
 
-    unsafe fn write_buffers(&self, draw_data: &DrawData) -> HResult<()> {
-        let mut vtx_resource = mem::MaybeUninit::zeroed();
-        let mut idx_resource = mem::MaybeUninit::zeroed();
-        hresult(self.context.Map(
-            self.vertex_buffer.as_raw().cast(),
-            0,
-            D3D11_MAP_WRITE_DISCARD,
-            0,
-            vtx_resource.as_mut_ptr(),
-        ))?;
-        if let e @ Err(_) = hresult(self.context.Map(
-            self.index_buffer.as_raw().cast(),
-            0,
-            D3D11_MAP_WRITE_DISCARD,
-            0,
-            idx_resource.as_mut_ptr(),
-        )) {
-            self.context.Unmap(self.vertex_buffer.as_raw().cast(), 0);
-            e?;
-        }
-        let vtx_resource = vtx_resource.assume_init();
-        let idx_resource = idx_resource.assume_init();
+    unsafe fn write_buffers(&self, draw_data: &DrawData) -> Result<()> {
+        let vtx_resource: D3D11_MAPPED_SUBRESOURCE = self.context.Map(self.vertex_buffer.get_buf(), 0, D3D11_MAP_WRITE_DISCARD, 0)?;
+        let idx_resource: D3D11_MAPPED_SUBRESOURCE = self.context.Map(self.index_buffer.get_buf(), 0, D3D11_MAP_WRITE_DISCARD, 0)?;
 
-        let mut vtx_dst = slice::from_raw_parts_mut(
+        let vtx_dst = slice::from_raw_parts_mut(
             vtx_resource.pData.cast::<DrawVert>(),
             draw_data.total_vtx_count as usize,
         );
-        let mut idx_dst = slice::from_raw_parts_mut(
+        let idx_dst = slice::from_raw_parts_mut(
             idx_resource.pData.cast::<DrawIdx>(),
             draw_data.total_idx_count as usize,
         );
-        for (vbuf, ibuf) in
-            draw_data.draw_lists().map(|draw_list| (draw_list.vtx_buffer(), draw_list.idx_buffer()))
-        {
-            vtx_dst[..vbuf.len()].copy_from_slice(vbuf);
-            idx_dst[..ibuf.len()].copy_from_slice(ibuf);
-            vtx_dst = &mut vtx_dst[vbuf.len()..];
-            idx_dst = &mut idx_dst[ibuf.len()..];
-        }
 
-        self.context.Unmap(self.vertex_buffer.as_raw().cast(), 0);
-        self.context.Unmap(self.index_buffer.as_raw().cast(), 0);
+        let mut vp = 0;
+        let mut ip = 0;
+        for (vbuf, ibuf) in draw_data.draw_lists().map(|draw_list| (draw_list.vtx_buffer(), draw_list.idx_buffer())) {
+            vtx_dst[vp..vp + vbuf.len()].copy_from_slice(vbuf);
+            idx_dst[ip..ip + ibuf.len()].copy_from_slice(ibuf);
+            vp += vbuf.len();
+            ip += ibuf.len();
+        }
```

Comment:
any reason for changing this to use indices again?

Reply:
Oddly this was crashing here when I was porting, but I think I was likely not creating a buffer correctly or something. Either way, I reverted the change